### ISSUE TITLE
In build mode, throw error if a model already exists in the output folder

### ISSUE THREAD

**Issue by hboisgon** [Deltares/hydromt#257]
When building a model in an output folder which already contains an existing model, I think hydromt should throw an error instead of building a model which may then partially conflicts with other existing model files from the already existing one.

This check could be done in self.set_root.
For now we do throw a warning for update mode "r+" but do not really do anything for build mode "w"

**Comment by DirkEilander** [comment]
If i read the code correctly a warning is logged when writing to an existing folder in write-only mode. For me a warning is sufficient here. We can add to the warning that it may result in a corrupt model.

This code snippet (from `Model.set_root`) throws a warning if `isdir(path) and (self._write and not self._read)` i.e. if path exist and in write-only mode.

```
if self._write:
    for name in self._FOLDERS:
        path = join(self._root, name)
        if not isdir(path):
            os.makedirs(path)
        elif not self._read:
            self.logger.warning(
                "Model dir already exists and "
                f"files might be overwritten: {path}."
            )
```

**Comment by hboisgon** [comment]
In update mode this is fine, but in build I really wonder if a warning is enough? The issue was raised by @xldeltares.

**Comment by DirkEilander** [comment]
No there is no warning in update (append) mode which we could add. An error will be annoying I think but perhaps that's behaviour we could ask more users their opinion on, e.g. via a teams polly?

**Comment by xldeltares** [comment]
Hi @hboisgon and @DirkEilander,  I myself find the warning insufficient because sometimes we work with different versions/scenarios of models. And I would not like my old folder to be overwritten without a confirmation/reminder before this action. But I agree a teams polly sounds like a good idea! I am also curious how others deal with this in practise.

**Comment by DirkEilander** [comment]
proposed solution: By default we don't overwrite an existing folder. A user can overrule with a `force_overwrite` flag in the hydromt build CLI method. We then pass a new `mode=w+` to Model.set_root(). For `mode=w` (default) we do not allow overwriting anymore.

**Comment by DirkEilander** [comment]
this is fixed in #238
